test/rules/avoid_implementing_value_types.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_implementing_value_types`
```

Comment:
There should be at least one test for each of the old and new mixin styles.

Reply:
Is the new mixin style supported? Are there other tests that currently do this?

Reply:
> Is the new mixin style supported?

Analyzer supports the new syntax and generates many of the required diagnostics. It should be far enough along to allow us to start testing in in lints. The front_end package currently only supports the syntax, but that shouldn't limit us.

> Are there other tests that currently do this?

No, we haven't yet gone through the existing lints to determine where we need to add tests for the new mixin syntax.

Reply:
Then can I push back on this requirement to land the lint? I don't know anything about the new mixin syntax, or how it would (or would not) affect this lint in a meaningful way.

Reply:
I do think we want a test(s) for the use of mixins, even if it's only the old style. For example:
```
class MixesInSize extends Object with Size { // LINT
}
```
A similar example using the new syntax would be:
```
mixin SizeMixin {
  @override
  bool operator==(Object o) => o is Size && o.inBytes == other.inBytes;

  @override
  int get hashCode => inBytes.hashCode;
}
class MixesInSizeMixin with SizeMixin { // LINT
}
```

Reply:
The first sample is invalid (`Size` declares a constructor), but I can make a sample that works

Reply:
Using the new `mixin` syntax causes the file to be flagged in the analyzer as invalid.